src/main/java/org/kuali/kra/subaward/service/impl/SubAwardServiceImpl.java

```diff
@@ -249,6 +251,17 @@ public SubAward getAmountInfo(SubAward subAward) {
         subAward.setTotalAmountReleased(totalAmountReleased);
         subAward.setTotalAvailableAmount(totalObligatedAmount.subtract(totalAmountReleased));
 
+        // BUKC-0145: Transactions on Financial Tab - Order switching (DFCT0011505)  - fixes issue with BUKC-0105
+        List<SubAwardAmountInfo> sortedlist = subAward.getSubAwardAmountInfoList();
+        if (!sortedlist.isEmpty() && !(sortedlist.get(sortedlist.size() - 1).getSubAwardAmountInfoId() == null)) {
+            Collections.sort(sortedlist, new Comparator<SubAwardAmountInfo>() {
+                public int compare(SubAwardAmountInfo o1, SubAwardAmountInfo o2) {
+                    return o1.getSubAwardAmountInfoId() - o2.getSubAwardAmountInfoId();

```

Comment:
I think the right place to check for nulls is in the comparator and address the case if one, or the second one or both are null.